### PR TITLE
Extract DiskUsageIntegTestCase

### PR DESCRIPTION
**Pull request by henningandersen**
Extracted DiskUsageIntegTestCase from DiskThresholdDeciderIT to allow
other tests to easily test functionality relying on disk usage.

Relates #65520

### CODE REVIEW

**Comment by henningandersen** [comment]
Build failure looks like #65541 

@elasticmachine update branch

**Comment by henningandersen** [comment]
@elasticmachine update branch

File: test/framework/src/main/java/org/elasticsearch/cluster/DiskUsageIntegTestCase.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.common.util.concurrent.ConcurrentCollections.newConcurrentMap;
+import static org.hamcrest.Matchers.anyOf;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * An integration test case that allows mocking the disk usage per node. Notice that only files count towards disk usage and translog and
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/cluster/DiskUsageIntegTestCase.java:60]:
```suggestion
 * An integration test case that allows mocking the disk usage per node. Notice that only Lucene files count towards disk usage and translog and
```

**Review by DaveCTurner — APPROVED** [review]
LGTM with one comment nit

**Comment by henningandersen** [comment]
@elasticmachine update branch

**Comment by henningandersen** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Comment by henningandersen** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Pull request merged**